library/math/numbersLine/src/hooks/useDraggableElementAction.ts

```diff
@@ -95,7 +95,8 @@ export const useDraggableElementAction = () => {
   };
 
   const updateDragElements = (elementId: string, newElement: IElement) => {
-    const newElements = dragElements.map((item: IElement) => (item.id === elementId ? newElement : item));
+    const dragElementsWithoutNewElement = dragElements.filter((item: IElement) => item.id != elementId);
```

Comment:
i dont know if it is good to work like that and not strange... to add a new element every change.... maby you should save zindex per element and change it when you touch to maxZIndex+1 its more clean and righf